Issue with generated code in TILMedia

In the currently tested public version of TILMedia, there is one remaining test case failing, namely [TestVLEFluidObjectFunctions](https://libraries.openmodelica.org/branches/master/TILMedia/files/TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions.err). It shows several warnings and one error during C compilation:

```
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:53:12: warning: implicit declaration of function \'TILMedia_redirectModelicaFormatMessage_wrapper\' is invalid in C99 [-Wimplicit-function-declaration]
  _x_ext = TILMedia_redirectModelicaFormatMessage_wrapper();
           ^
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:85:25: warning: implicit declaration of function \'TILMedia_createExternalObject\' is invalid in C99 [-Wimplicit-function-declaration]
  _externalObject_ext = TILMedia_createExternalObject(MMC_STRINGDATA(_objectType), MMC_STRINGDATA(_mediumName), _flags_ext, (const double*) _xi_c89, _nc_ext, _condensingIndex_ext, MMC_STRINGDATA(_instanceName));
                        ^
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:85:23: warning: incompatible integer to pointer conversion assigning to \'void *\' from \'int\' [-Wint-conversion]
  _externalObject_ext = TILMedia_createExternalObject(MMC_STRINGDATA(_objectType), MMC_STRINGDATA(_mediumName), _flags_ext, (const double*) _xi_c89, _nc_ext, _condensingIndex_ext, MMC_STRINGDATA(_instanceName));
                      ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:131:77: warning: passing \'const double *\' to parameter of type \'double *\' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  _h_ext = TILMedia_VLEFluidFunctions_specificEnthalpy_pTxi(_p_ext, _T_ext, (const double*) _xi_c89, MMC_STRINGDATA(_vleFluidName), _nc_ext);
                                                                            ^~~~~~~~~~~~~~~~~~~~~~~
./TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_includes.h:8:80: note: passing argument to parameter here
double TILMedia_VLEFluidFunctions_specificEnthalpy_pTxi(double, double, double*,const char*, int);
                                                                               ^
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:165:70: warning: passing \'const double *\' to parameter of type \'double *\' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  _T_ext = TILMedia_VLEFluid_Cached_temperature_phxi(_p_ext, _h_ext, (const double*) _xi_c89, _vleFluidPointer_ext);
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~
./TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_includes.h:4:73: note: passing argument to parameter here
double TILMedia_VLEFluid_Cached_temperature_phxi(double, double, double*, void*);
                                                                        ^
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:197:66: warning: passing \'const double *\' to parameter of type \'double *\' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  _d_ext = TILMedia_VLEFluid_Cached_density_phxi(_p_ext, _h_ext, (const double*) _xi_c89, _vleFluidPointer_ext);
                                                                 ^~~~~~~~~~~~~~~~~~~~~~~
./TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_includes.h:7:69: note: passing argument to parameter here
double TILMedia_VLEFluid_Cached_density_phxi(double, double, double*, void*);
                                                                    ^
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:229:79: warning: passing \'const double *\' to parameter of type \'double *\' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  _w_ext = TILMedia_VLEFluidObjectFunctions_speedOfSound_dTxi(_d_ext, _T_ext, (const double*) _xi_c89, _vleFluidPointer_ext);
                                                                              ^~~~~~~~~~~~~~~~~~~~~~~
./TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_includes.h:6:80: note: passing argument to parameter here
double TILMedia_VLEFluidObjectFunctions_speedOfSound_dTxi(double,double,double*,void*);
                                                                               ^
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:261:79: warning: passing \'const double *\' to parameter of type \'double *\' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  _w_ext = TILMedia_VLEFluidObjectFunctions_speedOfSound_phxi(_p_ext, _h_ext, (const double*) _xi_c89, _vleFluidPointer_ext);
                                                                              ^~~~~~~~~~~~~~~~~~~~~~~
./TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_includes.h:5:80: note: passing argument to parameter here
double TILMedia_VLEFluidObjectFunctions_speedOfSound_phxi(double,double,double*,void*);
                                                                               ^
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:293:70: warning: passing \'const double *\' to parameter of type \'double *\' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  _T_ext = TILMedia_VLEFluid_Cached_temperature_phxi(_p_ext, _h_ext, (const double*) _xi_c89, _vleFluidPointer_ext);
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~
./TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_includes.h:4:73: note: passing argument to parameter here
double TILMedia_VLEFluid_Cached_temperature_phxi(double, double, double*, void*);
                                                                        ^
TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.c:328:27: error: use of undeclared identifier \'_defaultMixingRatio\'
  index_alloc_real_array(&_defaultMixingRatio, &tmp6, &tmp3);
                          ^
9 warnings and 1 error generated.
make: *** [<builtin>: TILMedia_TILMedia.Testers.TestVLEFluidObjectFunctions_functions.o] Error 1
```
@mahge, would you mind having a look?

--- a/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
+++ b/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
@@ -3021,7 +3021,7 @@ template fillArrayFromRange(DAE.Type ty, Exp exp, DAE.ComponentRef cr, Context c
 match exp
 case RANGE(__) then
   let &preExp = buffer ""
-  let cref = contextArrayCref(cr, context)
+  let cref = contextCref(cr, context, &preExp, &varDecls, &auxFunction)
   let ty_str = expTypeArray(ty)
   let start_exp = daeExp(start, context, &preExp, &varDecls, &auxFunction)
   let stop_exp = daeExp(stop, context, &preExp, &varDecls, &auxFunction)
@@ -3042,7 +3042,7 @@ template indexedAssign(DAE.Exp lhs, String exp, Context context,
     let ispec = daeExpCrefIndexSpec(crefSubs(cr), context, &preExp, &varDecls, &auxFunction)
     match context
       case FUNCTION_CONTEXT(__) then
-        let cref = contextArrayCref(cr, context)
+        let cref = contextCref(crefStripLastSubs(cr), context, &preExp, &varDecls, &auxFunction)
         'indexed_assign_<%arrayType%>(<%exp%>, &<%cref%>, &<%ispec%>);'
       else
         let type = expTypeShort(aty)
@@ -5294,7 +5294,7 @@ template daeExpCrefRhsFunContext(Exp ecr, Context context, Text &preExp,
         case FUNCTION_CONTEXT(__) then
           // The array subscript denotes a slice
           // let &preExp += '/* daeExpCrefRhsFunContext SLICE(<%ExpressionDumpTpl.dumpExp(ecr,"\"")%>) preExp  */<%\n%>'
-          let arrName = contextArrayCref(cr, context)
+          let arrName = contextCref(crefStripLastSubs(cr), context, &preExp, &varDecls, &auxFunction)
           let arrayType = expTypeArray(ty)
           let tmp = tempDecl(arrayType, &varDecls)
           let spec1 = daeExpCrefIndexSpec(crefSubs(cr), context, &preExp, &varDecls, &auxFunction)

